Applications/Utils/FileConverter/NetCdfConverter.cpp

```diff
@@ -0,0 +1,696 @@
+                       bool has_info = false)
+{
+    while (true)
+    {
+        std::cout << request_str;
+        std::string str;
+        std::size_t val;
+        std::getline(std::cin, str);
+        checkExit(str);
+        if (has_info && str == "info")
+            return (max_val);
+        std::stringstream str_stream(str);
+        if (!(str_stream >> val))
+        {
+            std::size_t const error_val = (has_info) ? 2 : 0;
+            showErrorMessage(error_val);
+            continue;
+        }
+        if (val > max_val - 1)
+        {
+            showErrorMessage(1, max_val - 1);
+            continue;
+        }
+        return val;
+    }
+    return std::numeric_limits<std::size_t>::max();
+}
+
+NcVar getDimVar(NcFile const& dataset, NcVar const& var, std::size_t const dim)
+{
+    NcDim const& dim_obj = var.getDim(dim);
+    return dataset.getVar(dim_obj.getName());
+}
+
+std::vector<std::string> showArrays(NcFile const& dataset)
+{
+    std::size_t const n_vars(dataset.getDimCount());
+    std::cout << "The NetCDF file contains the following " << n_vars
+              << " arrays:\n\n";
+    std::cout << "\tIndex\tArray Name\t#Dimensions\n";
+    std::cout << "-------------------------------------------\n";
+    auto const& names = dataset.getVars();
+    std::vector<std::string> var_names;
+    for (auto [name, var] : names)
+    {
+        std::cout << "\t" << var_names.size() << "\t" << name << "\t("
+                  << var.getDimCount() << "D array)\n";
+        var_names.push_back(name);
+    }
+    std::cout << "\n";
+    return var_names;
+}
+
+bool showArraysDims(NcVar const& var)
+{
+    std::cout << "Data array \"" << var.getName()
+              << "\" contains the following dimensions:\n";
+    std::size_t const n_dims(var.getDimCount());
+    for (std::size_t i = 0; i < n_dims; ++i)
+        std::cout << "\t" << i << "\t" << var.getDim(i).getName() << "\t("
+                  << var.getDim(i).getSize() << " values)\n";
+    std::cout << "\n";
+    return true;
+}
+
+std::pair<double, double> getBoundaries(NcVar const& var)
+{
+    if ((var.getDimCount()) == 1)
+    {
+        double start, end;
+        std::size_t const size = var.getDim(0).getSize();
+        var.getVar({0}, {1}, &start);
+        var.getVar({size - 1}, {1}, &end);
+        return std::make_pair(start, end);
+    }
+    return std::make_pair(0, 0);
+}
+
+MathLib::Point3d getOrigin(NcFile const& dataset, NcVar const& var,
+                           std::array<std::size_t, 4> const& dim_idx_map,
+                           bool is_time_dep)
+{
+    std::size_t const temp_offset = (is_time_dep) ? 1 : 0;
+    MathLib::Point3d origin(std::array<double, 3>{{0, 0, 0}});
+    std::size_t const n_dims = var.getDimCount();
+    for (std::size_t i = temp_offset; i < n_dims; ++i)
+    {
+        NcVar const& dim = getDimVar(dataset, var, dim_idx_map[i]);
+        auto const bounds = getBoundaries(dim);
```

Comment:
What is the difference?

Reply:
The readability. What is bounds.first and bounds.second? Using structured bindings some meaningful variable names can be used.

Reply:
Alright ... a little help here ... getBoundaries() returns two boundary values. If bounds. first and bounds.second are too obscure, what do you propose I should call these?

Reply:
{min, max}, {left, right}, {start, end} whatever the meaning of the values is

Reply:
It's boundaries of a spatial dimension. If you look at the code, I'll get both values and then determine which one is smaller to calculate the offset. "left"/"right" is correct if these are in West/East direction but wrong in North/South direction. Min/max also doesn't work because I will switch them if the dataset is located in the southern hemisphere or if the dimension is flipped for some other reason (e.g. atmospheric vs subsurface data). It's just two boundary values.

Reply:
OK